pkgs/top-level/all-packages.nix

```diff
@@ -5416,6 +5416,8 @@ in
 
   sharutils = callPackage ../tools/archivers/sharutils { };
 
+  shelldap = perlPackages.Shelldap;
```

Comment:
does it need to be exposed at the root level?

Reply:
Well I think it needs as one should be able to install it directly without knowing it's installed from package? It is a command line tool before being a program written in in perl no?

Reply:
@tobiasBora you can install it via `perlPackages.Shelldap` as in `nix-env -f '<nixpkgs>' -iA perlPackages.Shelldap`

Reply:
I'd be inclined to expose it at the root, it isn't a perl lib, but a program.

Reply:
I'm with @grahamc on this.

Reply:
In could be also the reason for:
1. create it as top level only (without duplicating as ```perlPackages.Shelldap```)
2. crete in one nix file, not in perl-packages.nix, (I have a dream to see only CPAN packages in perl-packages.nix and no CPAN packages outside perl-packages.nix to simplify auto-update, and kicking the apps out of perl-packages.nix is a way to go)

Reply:
@volth : I think this make sense. If I do that were do you recommend to put the file? In nixpkgs/pkgs/applications/misc?

Reply:
I found `tools/misc/ldapvi`, guess it's still better.

Reply:
Is it better that way?

Reply:
@volth @grahamc thank you for clarifying this issue!